pennylane/templates/subroutines/qsvt.py

```diff
@@ -0,0 +1,193 @@
+# limitations under the License.
+"""
+Contains the QSVT template and qsvt wrapper function.
+"""
+import numpy as np
+import pennylane as qml
+from pennylane.queuing import QueuingManager
+from pennylane.ops import BlockEncode, PCPhase
+from pennylane.ops.op_math import adjoint
+from pennylane.operation import AnyWires, Operation
+
+
+def qsvt(A, phi_vect, wires):
+    """Performs the
+    `quantum singular value transformation <https://arxiv.org/abs/1806.01838>`__ circuit using
+    :class:`~.BlockEncode` and :class:`~.PCPhase`.
+
+    Args:
+        A (Array): the general :math:`(n \times m)` matrix to be encoded.
+        phi_vect (list[Float]): a list of angles by which to shift to obtain the desired polynomial.
+        wires (Iterable): the wires the template acts on.
+
+    .. details::
+        :title: Usage Details
+
+        .. code-block:: python
+
+            dev = qml.device('default.qubit', wires=2)
+            A = [[0.1, 0.2], [0.3, 0.4]]
+            lst_phis = [0.1, 0.2, 0.3]
+            @qml.qnode(dev)
+            def example_circuit(A):
+                qml.qsvt(A, lst_phis, wires=[0,1])
+                return qml.expval(qml.PauliZ(wires=0))
+
+            qml.draw(example_circuit)(A)
+    """
+    c, r = qml.math.shape(A)
+
+    U_A = qml.BlockEncode(A, wires=wires, do_queue=False)
+    lst_projectors = []
+    for idx, phi in enumerate(phi_vect):
+        if idx % 2 == 0:
+            lst_projectors.append(PCPhase(phi, dim=r, wires=wires, do_queue=False))
+        else:
+            lst_projectors.append(PCPhase(phi, dim=c, wires=wires, do_queue=False))
+
+    return QSVT(U_A, lst_projectors, wires=wires)
+
+
+class QSVT(Operation):
+    r"""Performs the 
+    `quantum singular value transformation <https://arxiv.org/abs/1806.01838>`__ circuit.
+
+    Given a circuit :math:`U(A)`, which block encodes the matrix :math:`A`, and a list of projector-controlled
+    phase shifts, this template applies the circuit for quantum singular value transformation.
+
+    .. math::
+
+        \begin{align}
+             U_{qsvt}(A, \phi) &=
+             \begin{bmatrix}
+                Poly^{SV}(A) & \cdot \\
+                \cdot & \cdot
+            \end{bmatrix}.
+        \end{align}
+
+    This circuit can be used to perform the standard quantum singular value transformation algorithm, consisting
+    of alternating block encoding and controlled phase shift operations.
+
+    Args:
+        U_A (Operator): the block encoding circuit, specified as a :class:`~.Operator` 
+            or quantum function.
+        lst_projectors (list[Operator] or list[Callable]): a list of projector-controlled phase
+            shifts that implement the desired polynomial.
+        wires (Iterable): the wires the template acts on.
+
+    Raises:
+        QuantumFunctionError: 
+
+    .. details::
+        :title: Usage Details
+
+        Consider the matrix corresponding to a rotation from an :class:`~.RX` gate:
```

Comment:
Should we also have an example for users to use a `Callable`? (not a blocker)

Reply:
Decided not to support callables for now.